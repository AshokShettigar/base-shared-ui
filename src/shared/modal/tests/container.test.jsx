import { shallow } from 'enzyme';

import Modal from '../container';

describe('Modal component', () => {
    it('should render component with default props', () => {
        const wrapper = shallow(<Modal />).shallow();

        expect(wrapper).toMatchSnapshot();
    });

    it('shouldn\'t render component', () => {
        const wrapper = shallow(<Modal rendered={false} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render opened modal', () => {
        const wrapper = shallow(<Modal isOpen />).shallow();

        expect(wrapper.state().isOpen).toBeTruthy();

        wrapper.setProps({ isOpen: false });
        wrapper.setProps({ isOpen: true });

        expect(wrapper.state().isOpen).toBeTruthy();

        expect(wrapper).toMatchSnapshot();
    });

    it('should render closed modal', () => {
        const wrapper = shallow(<Modal />).shallow();

        expect(wrapper.state().isOpen).toBeFalsy();

        wrapper.setProps({ isOpen: true });
        wrapper.setProps({ isOpen: false });

        expect(wrapper.state().isOpen).toBeFalsy();

        expect(wrapper).toMatchSnapshot();
    });

    it('shouldn\'t called onDismiss', () => {
        const handleDismiss = jest.fn();
        const wrapper = shallow(<Modal onDismiss={handleDismiss} />).shallow();

        wrapper.setProps({ isOpen: false });

        expect(handleDismiss).not.toHaveBeenCalled();
    });

    it('should render with classes', () => {
        const props = {
            className: 'className',
            overlayClassName: 'overlayClassName',
            contentClassName: 'contentClassName',
            bodyOpenClassName: 'bodyOpenClassName'
        };
        const wrapper = shallow(<Modal {...props} />).shallow();

        expect(wrapper).toMatchSnapshot();
    });

    it('should render with children', () => {
        const wrapper = shallow(<Modal>Modal Content</Modal>).shallow();

        expect(wrapper).toMatchSnapshot();
    });

    it('should render with position', () => {
        const wrapper = shallow(<Modal position="middle" />).shallow();

        expect(wrapper).toMatchSnapshot();
    });

    it('should render with sections', () => {
        const props = {
            title: <span>Title Content</span>,
            header: <span>Header Content</span>,
            sidebar: <span>Sidebar Content</span>,
            footer: <span>Footer Content</span>
        };
        const wrapper = shallow(<Modal {...props} />).shallow();

        expect(wrapper).toMatchSnapshot();
    });

    it('should render with sections', () => {
        const props = {
            onAfterOpen: jest.fn(),
            onDismiss: jest.fn()
        };
        const wrapper = shallow(<Modal {...props} />).shallow();

        expect(wrapper).toMatchSnapshot();
    });
});

