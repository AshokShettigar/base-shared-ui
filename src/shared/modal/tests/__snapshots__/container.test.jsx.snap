// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render closed modal 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="modal-open"
  className="content hide bottomSheet"
  closeTimeoutMS={300}
  contentLabel=""
  isOpen={false}
  onRequestClose={[Function]}
  overlayClassName="overlay modal-position-bottom"
  parentSelector={[Function]}
  portalClassName="root"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={false}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <ModalBodySection
    sidebar={null}
  />
</Modal>
`;

exports[`Modal component should render component with default props 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="modal-open"
  className="content hide bottomSheet"
  closeTimeoutMS={300}
  contentLabel=""
  isOpen={false}
  onRequestClose={[Function]}
  overlayClassName="overlay modal-position-bottom"
  parentSelector={[Function]}
  portalClassName="root"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={false}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <ModalBodySection
    sidebar={null}
  />
</Modal>
`;

exports[`Modal component should render opened modal 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="modal-open"
  className="content show bottomSheet"
  closeTimeoutMS={300}
  contentLabel=""
  isOpen={true}
  onRequestClose={[Function]}
  overlayClassName="overlay modal-position-bottom"
  parentSelector={[Function]}
  portalClassName="root"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={false}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <ModalBodySection
    sidebar={null}
  />
</Modal>
`;

exports[`Modal component should render with children 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="modal-open"
  className="content hide bottomSheet"
  closeTimeoutMS={300}
  contentLabel=""
  isOpen={false}
  onRequestClose={[Function]}
  overlayClassName="overlay modal-position-bottom"
  parentSelector={[Function]}
  portalClassName="root"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={false}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <ModalBodySection
    sidebar={null}
  >
    Modal Content
  </ModalBodySection>
</Modal>
`;

exports[`Modal component should render with classes 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="bodyOpenClassName"
  className="content contentClassName hide bottomSheet"
  closeTimeoutMS={300}
  contentLabel=""
  isOpen={false}
  onRequestClose={[Function]}
  overlayClassName="overlay modal-position-bottom overlayClassName"
  parentSelector={[Function]}
  portalClassName="root className"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={false}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <ModalBodySection
    sidebar={null}
  />
</Modal>
`;

exports[`Modal component should render with position 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="modal-open"
  className="content hide"
  closeTimeoutMS={300}
  contentLabel=""
  isOpen={false}
  onRequestClose={[Function]}
  overlayClassName="overlay modal-position-middle"
  parentSelector={[Function]}
  portalClassName="root"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={false}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <ModalBodySection
    sidebar={null}
  />
</Modal>
`;

exports[`Modal component should render with sections 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="modal-open"
  className="content hide bottomSheet"
  closeTimeoutMS={300}
  contentLabel=""
  isOpen={false}
  onRequestClose={[Function]}
  overlayClassName="overlay modal-position-bottom"
  parentSelector={[Function]}
  portalClassName="root"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={false}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <span>
    Title Content
  </span>
  <span>
    Header Content
  </span>
  <ModalBodySection
    sidebar={
      <span>
        Sidebar Content
      </span>
    }
  />
  <ModalFooterSection>
    <span>
      Footer Content
    </span>
  </ModalFooterSection>
</Modal>
`;

exports[`Modal component should render with sections 2`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="modal-open"
  className="content hide bottomSheet"
  closeTimeoutMS={300}
  contentLabel=""
  isOpen={false}
  onAfterOpen={[Function]}
  onRequestClose={[Function]}
  overlayClassName="overlay modal-position-bottom"
  parentSelector={[Function]}
  portalClassName="root"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={false}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <ModalBodySection
    sidebar={null}
  />
</Modal>
`;

exports[`Modal component shouldn't render component 1`] = `null`;
